modules/claims_api/lib/custom_error.rb

```diff
@@ -8,83 +8,41 @@ def initialize(error, claim, method)
       @method = method
     end
 
-    def build_error # rubocop:disable Metrics/MethodLength
-      get_status
-      get_code
+    def handle_errors
       get_source
-      if @error.is_a?(Faraday::ConnectionFailed) || @error.is_a?(Faraday::ParsingError) ||
-         @error.is_a?(Faraday::NilStatusError) || @error.is_a?(Faraday::TimeoutError) ||
-         @error.is_a?(::Common::Exceptions::BackendServiceException) ||
-         @error.is_a?(::Common::Exceptions::ExternalServerInternalServerError) ||
-         @error.is_a?(::Common::Exceptions::BadGateway) || @error.is_a?(Faraday::SSLError) ||
-         @error.is_a?(Faraday::ServerError)
-        errors = { errors: [{ 'title' => 'Service Exception',
-                              'key' => @source,
-                              'detail' => 'A re-tryable error has occurred, original_error: ' \
-                                          "#{@error}.", status: @status, code: @code }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::ServiceError, errors
-
-      elsif @error.is_a?(StandardError) || @error.is_a?(Faraday::BadRequestError) ||
-            @error.is_a?(Faraday::ConflictError) || @error.is_a?(Faraday::ForbiddenError) ||
-            @error.is_a?(Faraday::ProxyAuthError) || @error.is_a?(Faraday::ResourceNotFound) ||
-            @error.is_a?(Faraday::UnauthorizedError) || @error.is_a?(Faraday::UnprocessableEntityError) ||
-            @error.is_a?(Faraday::ClientError)
-
-        errors = { errors: [{ 'title' => 'Client error',
-                              'key' => @source,
-                              'detail' => 'A client exception has occurred, job will not be re-tried. ' \
-                                          "original_error: #{@error}.", status: '400', code: '400' }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::BadRequest, errors
+      case @error
+      when Faraday::ParsingError
+        raise_backend_exception('EVSS502', @error)
+      when ::Common::Exceptions::BackendServiceException
+        raise ::Common::Exceptions::Forbidden if @error&.original_status == 403
+
+        raise_backend_exception('EVSS400', @error) if @error&.original_status == 400
+        raise ::Common::Exceptions::Unauthorized if @error&.original_status == 401
       else
-        errors = { errors: [{ 'title' => 'Unknown error',
-                              'key' => @source,
-                              'detail' => 'An unknown error has occurred, and the custom_error file may ' \
-                                          "need to be modified. original_error: #{@error}.", status: @status,
-                              code: @code }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::ServiceError, errors
+        raise @error
```

Comment:
Referenced in https://github.com/department-of-veterans-affairs/vets-api/pull/15987/files#r1534403752

Reply:
@FonzMP I do not understand what you are telling me here.

Reply:
the referenced link was to the comment above which I think you resolved here: https://github.com/department-of-veterans-affairs/vets-api/pull/15987/commits/9fcca6daf2c09e836be825a2e6110e9fcdf87875